vortex-expr/src/expr.rs

```diff
@@ -8,24 +8,19 @@ use vortex::compute::{compare, Operator as ArrayOperator};
 use vortex::variants::StructArrayTrait;
 use vortex::{Array, IntoArray};
 use vortex_dtype::field::Field;
-use vortex_dtype::DType;
-use vortex_error::{vortex_bail, vortex_err, VortexExpect as _, VortexResult, VortexUnwrap};
+use vortex_error::{vortex_err, VortexExpect as _, VortexResult};
 use vortex_scalar::Scalar;
-use vortex_schema::Schema;
 
 use crate::Operator;
 
-const NON_PRIMITIVE_COST_ESTIMATE: usize = 64;
-const COLUMN_COST_MULTIPLIER: usize = 1024;
-
 pub trait VortexExpr: Debug + Send + Sync + PartialEq<dyn Any> {
     fn as_any(&self) -> &dyn Any;
 
     fn evaluate(&self, batch: &Array) -> VortexResult<Array>;
 
     fn references(&self) -> HashSet<Field>;
 
-    fn estimate_cost(&self, schema: &Schema) -> usize;
+    fn project(&self, projection: &[Field]) -> Option<Arc<dyn VortexExpr>>;
```

Comment:
I don't understand what this function is supposed to do, or why it is on this trait?

Reply:
The implementation is in #979. It's meant to project the filter into the requested column space or return empty if expression cannot be projected. This is very obscure way of naming but sort of made sense in my head. Not sure what a better name is